pymc3/tests/test_sg.py

```diff
@@ -0,0 +1,26 @@
+import numpy as np
+
+from pymc3.blocking import DictToArrayBijection
+from . import models
+from pymc3.step_methods.hmc.base_hmc import BaseHMC
+from pymc3.step_methods.hmc import SGFS
+import pymc3
+from pymc3.theanof import floatX
+from .checks import close_to
+from theano import tensor
+
+
+def example_model():
+    pm = pymc3
+    cT = tensor.vector()
+    batch_size = 10
+    total_size = 100
+    cT.tag.test_value = floatX(np.ones(batch_size))
+    with pm.Model() as model:
+        a = pm.Normal('a', mu=0, sd=1.0)
+        b = pm.Normal('b', mu=1.0, sd=1.0)
+        c = pm.Normal('c', mu=a + b, sd=1.0, observed=cT)
```

Comment:
@ferrine Any changes you would suggest, to the way I have used generator in the test file now ?